### PR TITLE
feat(WEBRTC-2079): Actions Unit Tests + Static Code Analysis fixes

### PR DESCRIPTION
**Pull request by Oliver-Zimmerman**
[WEBRTC-2079 - Actions Unit Tests + Static Code Analysis fixes.](https://telnyx.atlassian.net/browse/WEBRTC-2079)
---

This PR implements a Github Action to run Unit tests on Push, Pull Request and a cron job schedule. The scheduled unit tests act as a sort of BlackBox test for things like socket connections, etc. 

Additionally, we have modified some code as suggested by static code analysis (Formatting, unnecessary null checks, etc.).

Lastly, we have added an example folder which points to the general readme and links to the core folder of the repo which currently serves as our example. Included in all Readmes we have also added a pub dev and unit test badge to show those states. 

### Before changes
No automated tests or example folder for pub dev

### After changes
- Automated tests
- Example folder linking to core repo folder via readme
- Static code analysis changes
- Readme badges 

## ✋ Manual testing
1. View readmes and use badges (badges)
2. Run app as usual to test functionality (static code analysis changes)
3. Make sure tests are running on this PR (Automated tests)

### CODE REVIEW

File: packages/telnyx_webrtc/lib/peer/peer.dart [hunk 1]
```diff
@@ -348,14 +353,14 @@ class Peer {
     onDataChannel?.call(session, channel);
   }
 
-  Future<void> _createDataChannel(Session session,
+  /*Future<void> _createDataChannel(Session session,
```
Comment (KalpeshTalkar) [inline packages/telnyx_webrtc/lib/peer/peer.dart:356]: Hey @Oliver-Zimmerman, are these functions (`_createDataChannel` and `_closeSessionByPeerId`) commented intentionally?

**Review by KalpeshTalkar — APPROVED** [review]
Just a minor comment regarding the commented functions. Rest LGMT!

File: packages/telnyx_webrtc/lib/peer/peer.dart (hunk 1 above)
Comment (Oliver-Zimmerman) [reply packages/telnyx_webrtc/lib/peer/peer.dart:356]: @KalpeshTalkar Yeah, we may need them later for later features, however, static code analysis doesn't like uncalled methods and brings our pub score down. I am leaving them commented out instead for now!

Comment (KalpeshTalkar) [reply packages/telnyx_webrtc/lib/peer/peer.dart:356]: Got it. Thanks for the explanation :)

**Pull request merged**